modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -538,6 +538,45 @@ def reindex(self, axis, labels, **kwargs):
         return self.__constructor__(new_modin_frame)
 
     def reset_index(self, **kwargs):
+        if self._modin_frame._index_cache is None:
+
+            def _reset(df, *columns_idx, partition_idx):
+                _kw = dict(kwargs)
+                if len(columns_idx) > 1 and partition_idx == 0:
```

Comment:
could you please provide with an example of what's going on here?

I don't understand why we can't just do:
```python
return df.reset_index(drop=drop or partition_idx != 0)
```

Reply:
If the index does not have a name, but there is a column named `index`, then pandas gives `level_0` name to the new column. That is, knowledge of the full axis of the columns is required. Perhaps there is something similar for a multi-index. This implementation helps us get rid of the implementation details of this function in pandas. 

P.S. we have a test for the case: `test_reset_index`.